pulsar/error.go

```diff
@@ -121,6 +121,15 @@ const (
 	// fenced. Applications are now supposed to close it and create a
 	// new producer
 	ProducerFenced
+	// MaxConcurrentOperationsReached indicates that the maximum number of concurrent operations
+	// has been reached. This means that no additional operations can be started until some
+	// of the current operations complete.
+	MaxConcurrentOperationsReached
+	// TransactionCoordinatorNotEnabled indicates that the transaction coordinator is not enabled.
+	// This error is returned when an operation that requires the transaction coordinator is attempted
+	// but the transaction coordinator feature is not enabled in the system or the transaction coordinator
+	// has not initialized
+	TransactionCoordinatorNotEnabled
```

Comment:
These new error codes and related changes are also unrelated to this PR. You'd better open a new PR for it.

That's also a disadvantage of such a huge PR because unrelated changes can easily hide in the changes.

Reply:
This was introduced while I was fixing this issue. I agree to move it to a separate PR.